src/libponyc/ast/token.h

```diff
@@ -147,6 +147,7 @@ typedef enum token_id
   TK_IF,
   TK_IFDEF,
   TK_IFTYPE,
+  TK_IFTYPE_CLAUSE,
```

Comment:
In that case, I think I might go the other way with it, and consider this construct to be the `TK_IFTYPE`, and consider the outer construct to be a group of them; maybe a `TK_IFTYPE_SET`.